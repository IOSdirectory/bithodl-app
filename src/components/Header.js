import React, { Component } from 'react';
import { StyleSheet, View } from 'react-native';
import PropTypes from 'prop-types';
import { ifIphoneX } from 'react-native-iphone-x-helper';

import BackButton from './BackButton';
import HeaderAddress from './HeaderAddress';

const styles = StyleSheet.create({
  header: {
    backgroundColor: 'transparent',
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    height: 42,
    zIndex: 1000
  },
  backButton: {
    position: 'absolute',
    top: ifIphoneX(54, 30),
    left: 10
  }
});

export default class Header extends Component {
  render() {
    const { address, showBackButton, hideAddress } = this.props;

    return (
      <View style={styles.header}>
        {showBackButton ? <BackButton onPress={this.props.onBackPress} style={styles.backButton} iconStyle={this.props.backButtonIconStyle} /> : null }
        {address && !hideAddress ? (<HeaderAddress address={address} />) : null }
      </View>
    );
  }
}

Header.propTypes = {
  address: PropTypes.object,
  showBackButton: PropTypes.bool,
  onBackPress: PropTypes.func.isRequired,
  backButtonIconStyle: PropTypes.any,
  hideAddress: PropTypes.bool
};
