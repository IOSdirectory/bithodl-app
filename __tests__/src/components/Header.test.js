import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../../../src/components/Header';

jest.mock('../../../src/containers/BtcLabelContainer', () => 'BtcLabelContainer');

describe('Header', () => {
  let fakeAddress;

  beforeEach(() => {
    fakeAddress = {
      id: 'eadc2f85-8fed-4dfc-8934-63e5de6293e2',
      hash: '1JNHNJftiWu57UVpp2cjrNNHCeLwbNy3Zt',
      script: 'OP_RETURN',
      amount: 2000000,
      createdAt: 1471651200,
      unlockAt: 1482192000
    };
  });

  it('renders correctly with showBackButton set to false', () => {
    const tree = renderer.create(
      <Header showBackButton={false} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with showBackButton set to true', () => {
    const tree = renderer.create(
      <Header showBackButton={true} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe('when an address is specified', () => {
    it('renders correctly with hideAddress set to false', () => {
      const tree = renderer.create(
        <Header address={fakeAddress} hideAddress={false} />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('renders correctly with hideAddress set to true', () => {
      const tree = renderer.create(
        <Header address={fakeAddress} hideAddress={true} />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('when an address is not specified', () => {
    it('renders correctly with hideAddress set to false', () => {
      const tree = renderer.create(
        <Header hideAddress={false} />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('renders correctly with hideAddress set to true', () => {
      const tree = renderer.create(
        <Header hideAddress={true} />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
