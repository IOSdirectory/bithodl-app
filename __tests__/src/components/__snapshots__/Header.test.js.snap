// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly with showBackButton set to false 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "height": 42,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 1000,
    }
  }
/>
`;

exports[`Header renders correctly with showBackButton set to true 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "height": 42,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 1000,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 10,
        "opacity": 1,
        "position": "absolute",
        "top": 30,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#D49146",
              "fontSize": 32,
              "paddingLeft": 10,
              "paddingRight": 10,
            },
            undefined,
          ],
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`Header when an address is not specified renders correctly with hideAddress set to false 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "height": 42,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 1000,
    }
  }
/>
`;

exports[`Header when an address is not specified renders correctly with hideAddress set to true 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "height": 42,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 1000,
    }
  }
/>
`;

exports[`Header when an address is specified renders correctly with hideAddress set to false 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "height": 42,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 1000,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0.5)",
          "borderRadius": 40,
          "flexDirection": "row",
          "padding": 5,
          "paddingBottom": 4,
          "paddingTop": 4,
          "position": "absolute",
          "right": 20,
          "top": 33,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 16,
              "height": 32,
              "justifyContent": "center",
              "width": 32,
            },
            Object {
              "backgroundColor": "#D9E5C3",
            },
          ],
          Object {
            "borderRadius": 8,
            "height": 16,
            "marginRight": 8,
            "width": 16,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "left": 0,
            "position": "absolute",
            "top": 0,
          }
        }
      >
        <View
          borderWidth={0}
          color="#7FA835"
          progress={1}
          size={16}
          thickness={0.5}
        />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Array [
              Object {
                "fontSize": 11,
                "fontWeight": "600",
              },
              Object {
                "color": "#7FA835",
              },
            ],
            Object {
              "backgroundColor": "transparent",
              "fontSize": 7,
            },
          ]
        }
      >
        ✓
      </Text>
    </View>
    <BtcLabelContainer
      amount={2000000}
      style={
        Object {
          "color": "#504F63",
          "fontSize": 12,
          "fontWeight": "600",
          "marginRight": 3,
        }
      }
    />
  </View>
</View>
`;

exports[`Header when an address is specified renders correctly with hideAddress set to true 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "height": 42,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 1000,
    }
  }
/>
`;
